### PR TITLE
Add storybook for Gallery

### PR DESCRIPTION
**Pull request by freddyaboulton**
## Description

Adds some stories for the gallery component. One of our most used components and it has lots of styling options.

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests & Changelog

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

1. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`
  
1. Unless the pull request is labeled with the "no-changelog-update" label by a maintainer of the repo, all pull requests must update the changelog located in `CHANGELOG.md`:

Please add a brief summary of the change to the Upcoming Release section of the `CHANGELOG.md` file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

### CODE REVIEW

**Comment by abidlabs** [comment]
This is very cool @freddyaboulton but we shouldn't have to hardcode the parameters like this. As per the discussion this morning with @hannahblair we should be able to pull these programmatically from the docstrings, right?

**Comment by freddyaboulton** [comment]
My understanding is that it was somewhat a long way off. Don't think we should hold off on testing visual behavior until then. Especially since it's just a refactor of how the `argTypes` are specified - the stories themselves won't change.

**Comment by abidlabs** [comment]
That's fair, ok!

File: js/app/src/components/Gallery/Gallery.stories.svelte [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+	argTypes={{
+		label: {
+			control: "text",
+			description: "The gallery label",
+			name: "label",
+			value: "Gradio Button"
+		},
+		show_label: {
+			options: [true, false],
+			description: "Whether to show the gallery",
```
Comment (abidlabs) [inline js/app/src/components/Gallery/Gallery.stories.svelte:18]:
```suggestion
			description: "Whether to show the label",
```

**Comment by abidlabs** [comment]
Sweet! Just one typo

File: js/app/src/components/Gallery/Gallery.stories.svelte [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+		},
+		preview: {
+			options: [true, false],
+			description: "Whether to start the gallery in preview mode",
+			control: { type: "boolean" },
+			defaultValue: true
+		},
+		allow_preview: {
+			options: [true, false],
+			description: "Whether to allow a preview view in the gallery",
```
Comment (abidlabs) [inline js/app/src/components/Gallery/Gallery.stories.svelte:48]:
```suggestion
			description: "Whether to allow a preview mode in the gallery",
```

**Review by abidlabs — APPROVED** [review]
Sweet! Just a couple of wording nits

**Comment by dawoodkhan82** [comment]
Just looked at the story previews. Looks good to me.

**Review by dawoodkhan82 — APPROVED** [review]

**Comment by freddyaboulton** [comment]
Thanks @abidlabs and @dawoodkhan82 !

**Pull request merged**